### PR TITLE
oauth: Fix oauth code flow when user has not authorized the app (PROJQUAY-3648)

### PR DESCRIPTION
**Pull request by syed**
Fixes an issue where the code param is not passed to the app redirect
URI if the user has not authorized the app before

### CODE REVIEW

File: endpoints/web.py [hunk 1]
```diff
@@ -690,6 +690,7 @@ def request_authorization_code():
     redirect_uri = request.args.get("redirect_uri", None)
     scope = request.args.get("scope", None)
     state = request.args.get("state", None)
+    code = request.args.get("code", None)
```
Comment (kleesc) [inline endpoints/web.py:693]: The code shouldn't be included as part of the authorize request step of oauth. It's what we the Quay app would return as part of the uri when redirecting. The `code` is already be included, based on `oauth/provider.py`

**Review by kleesc — CHANGES_REQUESTED** [review]

**Review by kleesc — DISMISSED** [review]

File: requirements-dev.txt [hunk 2]
```diff
@@ -13,6 +13,7 @@ ipython
 mock==3.0.5
 mypy==0.910
 moto==2.0.1
+parameterized==0.8.1
```
Comment (kleesc) [inline requirements-dev.txt:16]: Is this required?

File: test/test_endpoints.py [hunk 3]
```diff
@@ -396,22 +397,26 @@ def test_redirect_to_namespace(self):
 
 
 class OAuthTestCase(EndpointTestCase):
-    def test_authorize_nologin(self):
+    @parameterized.expand(["token", "code"])
```
Comment (kleesc) [inline test/test_endpoints.py:400]: Could use pytest's existing parameterize?

Comment (syed) [reply test/test_endpoints.py:400]: can't use `pytest.parameterize` because the test class is based on `unittest.Testcase` which doesn't support it

**Review by kleesc — APPROVED** [review]

**Pull request merged**